akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -135,10 +135,26 @@ object ShardCoordinator {
   /**
    * The default implementation of [[ShardCoordinator.LeastShardAllocationStrategy]]
    * allocates new shards to the `ShardRegion` with least number of previously allocated shards.
-   * It picks shards for rebalancing handoff from the `ShardRegion` with most number of previously allocated shards.
+   *
+   * When a node is removed from the cluster the shards on that node will be started on the remaining nodes,
+   * evenly spread on the remaining nodes (by picking regions with least shards).
+   *
+   * When a node is added to the cluster the shards on the existing nodes will be rebalanced to the new node.
+   * It picks shards for rebalancing from the `ShardRegion` with most number of previously allocated shards.
    * They will then be allocated to the `ShardRegion` with least number of previously allocated shards,
    * i.e. new members in the cluster. There is a configurable threshold of how large the difference
-   * must be to begin the rebalancing. The number of ongoing rebalancing processes can be limited.
+   * must be to begin the rebalancing. The difference between number of shards in the region with most shards and
+   * the region with least shards must be greater than the `rebalanceThreshold` for the rebalance to occur.
+   *
+   * A `rebalanceThreshold` of 1 gives the best distribution and therefore typically the best choice.
+   * A higher threshold means that more shards can be rebalanced at the same time instead of one-by-one.
```

Comment:
As @johanandren correctly mentioned we have `maxSimultaneousRebalance` for this already so we might not have to use this setting for that?

Reply:
`maxSimultaneousRebalance` is not exactly the same thing. Rebalance is actually only about stopping shards, and then they will be started again on next access. That means that if there is low messaging traffic there can be new rebalance rounds before the shards have been allocated at the new place.

`maxSimultaneousRebalance` only limits that there are not too many stopping processes in flight at the same time. When a shard has been stopped (which is typically rather quick) it is not in progress anymore.

Note that the default rebalance tick is 10 seconds, so it's intentionally a rather slow process.